File: personal-service/src/main/java/com/inkasalud/personalservice/service/PersonalService.java
package com.inkasalud.personalservice.service;

import com.inkasalud.personalservice.dto.PersonalDto;

import java.util.List;

public interface PersonalService {
    List<PersonalDto> listar();
    PersonalDto obtener(Long id);
    PersonalDto crear(PersonalDto dto);
    PersonalDto actualizar(Long id, PersonalDto dto);
    void eliminar(Long id);
}

File: frontend/src/app/services/personal.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { Personal } from '../models/personal.model';

@Injectable({ providedIn: 'root' })
export class PersonalService {
  private http = inject(HttpClient);
  private apiUrl = `${environment.apiUrl}/personal`;

  listar(): Observable<Personal[]> {
    return this.http.get<Personal[]>(this.apiUrl);
  }

  obtener(id: number): Observable<Personal> {
    return this.http.get<Personal>(`${this.apiUrl}/${id}`);
  }

  crear(personal: Personal): Observable<Personal> {
    return this.http.post<Personal>(this.apiUrl, personal);
  }

  actualizar(id: number, personal: Personal): Observable<Personal> {
    return this.http.put<Personal>(`${this.apiUrl}/${id}`, personal);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
